cli: TestUnavailableZip failed

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13098296?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13098296?buildTab=artifacts#/) on master @ [96374f0c0c4758c93403631ce49c1c7bc93ef42e](https://github.com/cockroachdb/cockroach/commits/96374f0c0c4758c93403631ce49c1c7bc93ef42e):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip2926621408
[debug --host=127.0.0.1:43879 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:497: -- test log scope end --
--- FAIL: TestUnavailableZip (76.11s)
=== RUN   TestUnavailableZip/server_2
[debug --host=127.217.235.249:34023 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    testutils.go:122: panic: runtime error: invalid memory address or nil pointer dereference
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip2926621408
    --- FAIL: TestUnavailableZip/server_2 (38.33s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/obs-inf-prs @cockroachdb/server
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34527

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13129408?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13129408?buildTab=artifacts#/) on master @ [52329bc384068b9ea3e0c0e1100397f537dd2b89](https://github.com/cockroachdb/cockroach/commits/52329bc384068b9ea3e0c0e1100397f537dd2b89):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip1849997931
    zip_test.go:497: -- test log scope end --
--- FAIL: TestUnavailableZip (76.14s)
=== RUN   TestUnavailableZip/server_2
[debug --host=127.84.182.11:34193 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    testutils.go:122: panic: runtime error: invalid memory address or nil pointer dereference
test logs left over in: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip1849997931
    --- FAIL: TestUnavailableZip/server_2 (38.82s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13160858?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13160858?buildTab=artifacts#/) on master @ [512fc0e95bbd6690740c6e77b4efce6a4edd58b2](https://github.com/cockroachdb/cockroach/commits/512fc0e95bbd6690740c6e77b4efce6a4edd58b2):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip3739867215
    zip_test.go:497: -- test log scope end --
--- FAIL: TestUnavailableZip (76.29s)
=== RUN   TestUnavailableZip/server_1
[debug --host=127.0.0.1:35915 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
[debug --host=127.225.206.203:36453 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    testutils.go:122: panic: runtime error: invalid memory address or nil pointer dereference
    --- FAIL: TestUnavailableZip/server_1 (31.74s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13197514?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13197514?buildTab=artifacts#/) on master @ [6556ee673bfc8938511981ea7892d00d75448cfa](https://github.com/cockroachdb/cockroach/commits/6556ee673bfc8938511981ea7892d00d75448cfa):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2d3a1724ffe0f36d7c3bd890e201d037/logTestUnavailableZip2550299971
[debug --host=127.0.0.1:34527 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    zip_test.go:497: -- test log scope end --
--- FAIL: TestUnavailableZip (74.52s)
=== RUN   TestUnavailableZip/server_2
[debug --host=127.185.28.49:46545 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    testutils.go:122: panic: runtime error: invalid memory address or nil pointer dereference
    --- FAIL: TestUnavailableZip/server_2 (38.35s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

cli.TestUnavailableZip [failed](https://tanzanite.cluster.engflow.com/invocations/default/90dc3ee9-d238-4be6-b978-26a59fd35b82?testReportRun=14&testReportShard=4&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on master @ [d2d974202781321ddb4fb1115c68fe7566ab9d06](https://github.com/cockroachdb/cockroach/commits/d2d974202781321ddb4fb1115c68fe7566ab9d06):


```
=== RUN   TestUnavailableZip
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestUnavailableZip469557031
    zip_test.go:497: -- test log scope end --
test logs left over in: outputs.zip/logTestUnavailableZip469557031
--- FAIL: TestUnavailableZip (72.46s)
=== RUN   TestUnavailableZip/server_1
[debug --host=127.0.0.1:37353 --insecure=true zip --concurrency=1 --cpu-profile-duration=0 /dev/null --timeout=.5s]
    testutils.go:122: panic: runtime error: invalid memory address or nil pointer dereference
    --- FAIL: TestUnavailableZip/server_1 (30.73s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=14</code>
 - <code>shard=4</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestUnavailableZip.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -434,15 +434,12 @@ func TestUnavailableZip(t *testing.T) {
 
 	t.Run("server 1", func(t *testing.T) {
 		c := TestCLI{
-			t:        t,
 			Server:   tc.Server(0),
 			Insecure: true,
 		}
 
 		out, err := c.RunWithCapture(debugZipCommand)
-		if err != nil {
-			c.fail(err)
-		}
+		require.NoError(t, err)
 
 		// Assert debug zip output for cluster, node 1, node 2, node 3.
 		assert.NotEmpty(t, out)
@@ -470,15 +467,12 @@ func TestUnavailableZip(t *testing.T) {
 	t.Run("server 2", func(t *testing.T) {
 		// Run debug zip against node 2.
 		c := TestCLI{
-			t:        t,
 			Server:   tc.Server(1),
 			Insecure: true,
 		}
 
 		out, err := c.RunWithCapture(debugZipCommand)
-		if err != nil {
-			c.fail(err)
-		}
+		require.NoError(t, err)
 
 		// Assert debug zip output for cluster, node 2.
 		assert.NotEmpty(t, out)

